sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/models/HttpRequestContext.java

```diff
@@ -175,8 +177,17 @@ public Map<String, String> getHeaders() {
      * @param key the header key.
      * @param value the header value.
      */
-    public void addHeader(String key, String value) {
-        headers.put(key, value);
+    public void addHeaderParam(String key, String value) {
+        headerParams.put(key, value);
+    }
+
+    /**
+     * Sets the static headers.
+     *
+     * @param headers the static headers
+     */
+    public void setHeaders(String[] headers) {
```

Comment:
This can be confusing. By calling `setHeaders()`, it is only setting the static header collection and not overwriting the headers added through `addHeaderParam()` but `set*` generally means overwrite existing value.